### ISSUE TITLE
core dumping when probing type info

### ISSUE THREAD

**Issue by joshuareagan** [urbit/arvo#493]
I can reliably produce a core dump as follows.  In `:dojo`:

```
> =c -:!>(:-(2 17))
> +12:c
%atom
> +13:c

bail: fail
bailing out
Aborted (core dumped)
```

**Comment by ohAitch** [comment]
Hmm, is this fundamentally different from e.g. (reap 50.000 ~)? I guess
that hangs forever but doesn't actually bail.

My guess is this is a pretty printing crash, which is run outside a
virtualization for reasons only @cgyarvin knows

**Comment by joshuareagan** [comment]
Still getting a bail fail on this, but instead of "Abort (core dumped)" I'm returned to the dojo.

**Comment by joemfb** [comment]
The change was in urbit/urbit#1109, more context in #69. I don't think the bail:fail is a bug -- it could presumably be a bail:exit, but the compiler jets are cautious.

**Comment by joshuareagan** [comment]
You're right, this is fine.

**Comment by joshuareagan** [comment]
having second thoughts ---  This runs fine:

```
> =c -:!>([2 17])
> +2:c
%cell
> +3:c
[#t/@ud #t/@ud]
> +6:c
#t/@ud
```
But then the following bails:

```
+12:c
*bail*
```

We get the same bail if we put `+6:c` in `d` and try to take the head:

```
> =d +6:c

> +2:d
*bail*
```

But if we cast to a noun first it works fine:

```
> =e `*`+6:c

> +2:e
1.836.020.833
```

It seems to me that this conservative bail, while harmless to my urbit, is a pain for anyone who wants to explore type data.

**Comment by joshuareagan** [comment]
same outcome with `+12.c`

**Comment by joemfb** [comment]
You need to prefix with 5 `?` to activate the new printer.

**Comment by joshuareagan** [comment]
oops, I didn't do `????`...

```
> ???? +12.c
%atom
````

**Comment by joshuareagan** [comment]
```
> ????? +12.c
  
$@  $?  %hint
        %fork
        %face
        %noun
        ~
        %atom
        %void
        %core
        %cell
        %hold
    ==
$^  $:  n=(pair @tas (unit hoon))
        l=(map @tas (unit hoon))
        r=(map @tas (unit hoon))
    ==
$%  $:  %hold
        $$  $aynres
        ++  aynres
          p=type
        --
        (pair (list limb) hoon)
    ==
    $:  %core
        $$  $aynres
        ++  aynres
          p=type
        --
        q=coil
    ==
    [%fork p=(set type)]
    [%atom p=@tas q=(unit atom)]
    $:  %cell
        $$  $aynres
        ++  aynres
          p=type
        --
        q=type
    ==
    [%hint p=(pair type note) q=type]
    [%face p=$@(@tas tune) q=type]
==
%atom
```

**Comment by jtobin** [comment]
This one appears to be resolved as of `v0.8.0.rc.1`.
